griptape/events/event_listener.py

```diff
@@ -1,11 +1,20 @@
+from __future__ import annotations
+import uuid
 from typing import Callable, Optional, Type, Any
-from attrs import define, field
+from attrs import define, field, Factory
 from .base_event import BaseEvent
 
 
 @define
 class EventListener:
+    id: str = field(default=Factory(lambda: uuid.uuid4().hex), kw_only=True)
     handler: Callable[[BaseEvent], Any] = field()
     event_types: Optional[list[Type[BaseEvent]]] = field(
         default=None, kw_only=True
     )
+
+    def __eq__(self, other: EventListener) -> bool:
```

Comment:
I think you can just use the built-in `id()` method like this `id(self) == id(other)`, so no need to add `EventListener.id`.